### PR TITLE
Fix one error in travis file, get rid of warning and remove JSONNetworkTests

### PR DESCRIPTION
**Pull request by kchitalia**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ script:
    - xcodebuild -workspace SalesforceMobileSDK.xcworkspace -scheme NoteSync CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO | xcpretty --color --simple
    - xcodebuild -workspace SalesforceMobileSDK.xcworkspace -scheme SmartSyncExplorerHybrid CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO | xcpretty --color --simple
    # run tests that don't need a host app
-   - xcodebuild test -workspace SalesforceMobileSDK.xcworkspace -scheme SalesforceNetwork -sdk $SDK -destination "platform=iOS Simulator,name=$NAME,OS=$OS" | xcpretty --color --simple --test
+   - xcodebuild test -workspace SalesforceMobileSDK.xcworkspace -scheme SalesforceNetworkTests -sdk $SDK -destination "platform=iOS Simulator,name=$NAME,OS=$OS" | xcpretty --color --simple --test
```
Comment (kchitalia) [inline .travis.yml]: This should be SalesforcenetworkTests.

File: libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Utilities/CSFISO8601DateFormatter.m [hunk 2]
```diff
@@ -154,7 +154,7 @@ - (NSDateComponents *) dateComponentsFromString:(NSString *)string timeZone:(out
     
 	NSUInteger
 		//Date
-		year,
+		year = 0U,
```
Comment (kchitalia) [inline libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Utilities/CSFISO8601DateFormatter.m]: This gets rid of the warning "year might be null".

File: libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkTests/JSONNetworkTests.m [hunk 3]
```diff
@@ -1,62 +0,0 @@
-/*
```
Comment (kchitalia) [inline libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkTests/JSONNetworkTests.m]: Removing this file as we dont have any tests here. At this point I dont see any value in keeping this around. We can add it in later as and when needed.

**Comment by kchitalia** [comment]
There are still a few tests that are failing in the SalesforceNetwork project. I am looking into fixing them.

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**